client/src/get-ticket/components/home.tsx

```diff
@@ -1,14 +1,45 @@
-import { useState } from 'react'
+import { useEffect } from 'react'
 import Button from 'react-bootstrap/Button';
 import '../get-ticket.css';
 import { useNavigate } from 'react-router-dom';
+import API from '../../API/API.tsx';
 
-function Home() {
+function Home(props: any) {
 
+    const dirty = props.dirty;
+    const setDirty = props.setDirty;
+
+    useEffect(() => {
+        setDirty(true);
+      }, [])
+
+    // get all the services
```

Comment:
This is redundant.
Since you already defined the "services" state in the App that shared between two components.
I recommend to move this useEffect() logic into the App and use it once.
So, please remove it from Ticket and Home and put it in the App then drill down the result into child components.